pkg/doc/verifiable/credential.go

```diff
@@ -796,12 +826,34 @@ func newCredential(raw *rawCredential) (*Credential, error) {
 		Status:         raw.Status,
 		Schemas:        schemas,
 		Evidence:       raw.Evidence,
-		TermsOfUse:     raw.TermsOfUse,
-		RefreshService: raw.RefreshService,
+		TermsOfUse:     termsOfUse,
+		RefreshService: refreshService,
 		CustomFields:   raw.CustomFields,
 	}, nil
 }
 
+func decodeTypedID(bytes json.RawMessage) ([]TypedID, error) {
+	if len(bytes) == 0 {
+		return nil, nil
+	}
+
+	var singleTypedID TypedID
+
+	err := json.Unmarshal(bytes, &singleTypedID)
+	if err == nil {
+		return []TypedID{singleTypedID}, nil
+	}
+
+	var composedTypedID []TypedID
+
+	err = json.Unmarshal(bytes, &composedTypedID)
+	if err == nil {
+		return composedTypedID, nil
+	}
```

Comment:
Just wondering what's going on here (where you unmarshal to a TypedID and if that fails, you try unmarshalling to an array of TypedID). Is that because you don't know if the JSON will be an array or a single value?

Reply:
Yes, exactly. It can be defined either as a single value or as an array. It's a common thing for VC data model (unfortunately).

Reply:
Gotcha - just wanted to confirm :)